azavea/raster-foundry
# Add tile authentication

## Overview

This pull request adds support for specifying a token query parameter to tile requests and requires authentication by default to make tile requests. An environment variable is exposed such that in development if it is desired authentication can be bypassed; however, by default authentication is always enabled.

### Checklist

~- [ ] Styleguide updated, if necessary~
~- [ ] Swagger specification updated, if necessary~

## Testing Instructions
 - Start server and front-end
 - Open developer tools and browse to color correction
 - Observe that requests send token as query parameter
 - Copy a request for a tile into a new browser tab
 - Remove token query parameter and make request, observe request fails
 - Stop server and add environment variable to disable tile authentication
![set-env](https://cloud.githubusercontent.com/assets/898060/22065558/cd429868-dd56-11e6-804e-5d144770d36f.png)
 - Make request again for a tile without query parameter, request should succeed
